ecl/hql/hqlexpr.hpp

```diff
@@ -913,6 +913,8 @@ class HQL_API HqlParseContext
     inline void setCheckSimpleDef() { checkSimpleDef = true; }
     inline void setRegenerateCache() { regenerateCache = true; }
     inline void setIgnoreCache() { ignoreCache = true; }
+    void setExcludeFromCache(const char *regex) { excludeAttribsFromCache.assign(regex, std::regex_constants::icase); };
+    bool excludeFromCache(const char *fullname) { return std::regex_match(fullname, excludeAttribsFromCache); };
```

Comment:
How efficient is this if the regex has never been set?

Reply:
It's at least 2% slower for aslow1.  I'll add code to check before using std::regex.